src/crimson/osd/osd_operations/snaptrim_event.cc

```diff
@@ -252,10 +253,18 @@ SnapTrimObjSubEvent::remove_clone(
   delta_stats.num_object_clones--;
 
   obc->obs.exists = false;
-  obc->ssc->snapset.clones.erase(p);
-  obc->ssc->snapset.clone_overlap.erase(last);
-  obc->ssc->snapset.clone_size.erase(last);
-  obc->ssc->snapset.clone_snaps.erase(last);
+  head_obc->ssc->snapset.clones.erase(p);
+  head_obc->ssc->snapset.clone_overlap.erase(last);
+  head_obc->ssc->snapset.clone_size.erase(last);
+  head_obc->ssc->snapset.clone_snaps.erase(last);
+
+  ceph::bufferlist bss;
+  encode(head_obc->ssc->snapset, bss);
+  txn.setattr(
+    pg->get_collection_ref()->get_cid(),
+    ghobject_t{head_obc->obs.oi.soid},
+    SS_ATTR,
+    bss);
```

Comment:
IIUC, we will set the SS_ATTR in `update_head` later on.
If that is the case we can omit this section.

Reply:
> nit, I would also suggest splitting the changes here into 2 commits. First one is getting the ssc from the head obc while the second one is encoding and setting the head obc updated snapset attributes where needed.

You are right. And it seems that all "SS_ATTR" encoding added in this PR is not necessary. I've removed all of them, all the rests are "getting ssc from head obc". Please take a look, again, thanks:-)